ui/src/app/utils/subscribable.tsx

```diff
@@ -0,0 +1,43 @@
+import * as fp from 'lodash/fp';
+
+export interface Subscribable<T> {
+  subscribe: (Function) => { unsubscribe: () => void};
+  next: (newValue: T, oldValue: T) => void;
+}
+
+export interface Atom<T> {
+  get: () => T;
+  set: (value: T) => void;
+  subscribe: Subscribable<T>['subscribe'];
+  reset: () => void;
+}
+
+export function subscribable<T>(): Subscribable<T> {
+  let subscribers = [];
+  return {
+    subscribe: fn => {
+      subscribers = fp.concat(subscribers, [fn]);
+      return {
+        unsubscribe: () => {
+          subscribers = fp.without([fn], subscribers);
+        }
+      };
+    },
+    next: (...args) => {
+      fp.forEach(fn => fn(...args), subscribers);
```

Comment:
This immediately throws if any of the subscribers throw. If you keep this synchronous per above, I would have some try/catch protection around each callback, since this is potentially a large list of mixed subscribers - an error in one should not be able to take down the whole subscribable.